LabDrivers/SIM900.py

```diff
@@ -24,99 +25,563 @@
+                }
+            }
+        },
+        'Averaging Time':{
+            'type':'int',
+            'range':[0,10000],
+            'unit':'ms'
+        }
+    }
+    return ret
+
+properties = {
+    'Voltage': {
+        'type':'float',
+        'range':[-120, 120],
+        'unit':'V'
+    },
+    'Channel': {
+        'type':'int',
+        'range':[2, 15]
+    },
+    'Output': {
+        'type':'bool',
+        'range':True
+    },
+}
+properties = {}
+# set the properties
+if SIM910 != 0:
+    properties['SIM910'] = {
+        'type':'multi',
+        'range':create_SIM910_range(),
+        'properties':create_SIM910_properties_obj()
+    }
+if SIM928 != 0:
+    properties['SIM928'] = {
+        'type':'multi',
+        'range':create_SIM928_range(),
+        'properties':create_SIM928_properties_obj()
+    }
+if SIM980 != 0:
+    properties['SIM980'] = {
+        'type':'multi',
+        'range':create_SIM980_range(),
+        'properties':create_SIM980_properties_obj()
+    }
+
+
+READ_BITS = 64
+NAME = 'SIM900'
 INTERFACE = Tool.INTF_SERIAL
 
 
 class Instrument(Tool.MeasInstr):
 
     def __init__(self, resource_name, debug=False, **kwargs):
-        super(Instrument, self).__init__(resource_name, 'SIM900', debug, interface=INTERFACE, baud_rate=9600,
-                                         term_chars="\n", timeout=2, bytesize=8, parity='N', stopbits=1, xonxoff=False, dsrdtr=False, **kwargs)
-
-    def ask_channel(self, chan_num, msg):
-        #        print 'SEND %i,"%s"'%(chan_num,msg)
-        self.write('CONN %i,"xyz"' % chan_num)
-        answer = self.ask('%s' % (msg))
-        i.write("xyz")
+        super(Instrument, self).__init__(resource_name, NAME, debug, interface=INTERFACE, baud_rate=9600,
+                                         term_chars="\n".encode(), timeout=2, bytesize=8, parity='N', stopbits=1, xonxoff=False, dsrdtr=False, **kwargs)
+        self.connections = {}
+        device_temp = {}
+        self.ports = ['1','2','3','4','5','6','7','8']#,'A','B','C','D']
+        print("Identifying devices")
+        self.connected_devices = {}
+        for port in self.ports:
+            resp = self.ask_channel(port,"*IDN?")
+            if resp == '' or resp == None:
+                print("There is no device at port "+port)
+            else:
+                self.connections[port] = resp.split(',')[1]
+                if self.connections[port] in self.connected_devices.keys():
+                    self.connected_devices[self.connections[port]].append(port)
+                else:
+                    self.connected_devices[self.connections[port]] = [port]
+                print(self.connections[port]+" detected on port "+port)
+                device_temp[self.connections[port]] = True
+        self.devices = device_temp.keys()
+
+        try:
+            if 'SIM980' in self.devices:
+                #print(self.connected_devices)
```

Comment:
To avoid doing this, you could use logging. It allows you to print to the console with different levels: logging.debug(), logging.info(), logging.warning(), logging.error(). You can control how much display you get by setting a variable (you can even set the level from the GUI)